### PR TITLE
Fix copying "dev-404-page.js" to the cache folder on Windows

### LINKED ISSUE #2819: Since adding a static asset, HMR fails and development mode stops working
Using Windows 10 x64 with Node 8, I encounter the issue below quite often while trying to change a file in development mode (`gatsby develop`). I'm not sure about the cause, but the issue started when I added an asset to the project's `/static` directory.

```
 I  Your site is running at http://localhost:8000
 I  Your graphql debugger is running at http://localhost:8000/___graphql
info changed file at C:\Development\Projects\mvk-web\src\pages\gallery.jsx
 WAIT  Compiling...                                                                 00:16:17
error Plugin dev-404-page returned an error


  Error: EPERM: operation not permitted, open 'C:\Development\Projects\mvk-web\.cache\dev-40  4-page.js'

  - copy-file-sync.js:23 copyFileSync
    [mvk-web]/[fs-extra]/lib/copy-sync/copy-file-sync.js:23:18

  - copy-sync.js:43 Object.copySync
    [mvk-web]/[fs-extra]/lib/copy-sync/copy-sync.js:43:5

  - gatsby-node.js:34 _callee$
    [mvk-web]/[gatsby]/dist/internal-plugins/dev-404-page/gatsby-node.js:34:18

  - gatsby-node.js:51 Object.createPages
    [mvk-web]/[gatsby]/dist/internal-plugins/dev-404-page/gatsby-node.js:51:18

  - api-runner-node.js:104 runAPI
    [mvk-web]/[gatsby]/dist/utils/api-runner-node.js:104:36

  - api-runner-node.js:178
    [mvk-web]/[gatsby]/dist/utils/api-runner-node.js:178:33

  - map.js:27
    [mvk-web]/[gatsby]/[async]/internal/map.js:27:9

  - eachOfLimit.js:64 replenish
    [mvk-web]/[gatsby]/[async]/internal/eachOfLimit.js:64:17


 ERROR  Failed to compile with 1 errors                                             00:16:18
 error  in ./.cache/dev-404-page.js

Module build failed: Error: ENOENT: no such file or directory, open 'C:\Development\Projects\mvk-web\.cache\dev-404-page.js'

 @ ./.cache/sync-requires.js 11:53-123

info changed file at C:\Development\Projects\mvk-web\src\pages\gallery.jsx
 WAIT  Compiling...                                                                 00:16:22
error There was a problem reading the file: C:/Development/Projects/mvk-web/.cache/dev-404-page.js


  Error: ENOENT: no such file or directory, open 'C:\Development\Projects\mvk-web\.cache\dev  -404-page.js'

 ERROR  Failed to compile with 1 errors                                             00:16:23
 error  in ./.cache/dev-404-page.js

Module build failed: Error: ENOENT: no such file or directory, open 'C:\Development\Projects\mvk-web\.cache\dev-404-page.js'

 @ ./.cache/sync-requires.js 11:53-123

Done in 94.08s.
```

### PR DESCRIPTION
**Pull request by kripod**
Resolves #2819

### CODE REVIEW

File: packages/gatsby/src/bootstrap/load-plugins.js [hunk 1]
```diff
@@ -133,6 +133,9 @@ module.exports = async (config = {}) => {
   }
 
   // Add internal plugins
+  plugins.push(
+    processPlugin(path.join(__dirname, `../internal-plugins/dev-404-page`))
```
Comment (KyleAMathews) [inline packages/gatsby/src/bootstrap/load-plugins.js]: This was related to the fix?

Comment (kripod) [reply packages/gatsby/src/bootstrap/load-plugins.js]: I'm not quite sure, but examining the code lead me to the conclusion that the plugin loading order might be relevant.

Comment (KyleAMathews) [reply packages/gatsby/src/bootstrap/load-plugins.js]: Perhaps, it would make the copying run earlier which might matter for some reason haha 👍

**Comment by KyleAMathews** [comment]
The snapshot you changed is failing on TravisCI for some reason.

**Comment by kripod** [comment]
@KyleAMathews It turns out that content digests generated on my system (Windows 10 x64) are not equal to the ones generated by Travis CI's systems. I reverted the change caused by updating every snapshot automatically.

**Comment by KyleAMathews** [comment]
Hmmm that's not good. Oh probably because we're including file paths from plugins in the snapshots... hmmm didn't realize this was happening. Probably we should delete paths from the plugin data before snapshotting them so tests work cross-os.

Anyways, merging this now since Travis is happy but would love a PR for the ^^^ fix if you or someone else has time to look at it.

**Pull request merged**